sites/partners/lib/formatApplicationData.ts

```diff
@@ -0,0 +1,174 @@
+
+  const submissionDate: Date | null = (() => {
+    // rename default (wrong property names)
+    const { birthDay: submissionDay, birthMonth: submissionMonth, birthYear: submissionYear } =
+      data.dateSubmitted || {}
+    const { hours, minutes = 0, seconds = 0, period } = data.timeSubmitted || {}
+
+    if (!submissionDay || !submissionMonth || !submissionYear) return null
+
+    const date = new Date()
+
+    date.setUTCDate(parseInt(submissionDay))
+    date.setUTCMonth(parseInt(submissionMonth) - 1)
+    date.setUTCFullYear(parseInt(submissionYear))
+
+    if (hours && minutes && seconds && period) {
+      const hourNumber = parseInt(hours, 10)
+      const hour24Clock = period === "am" ? hourNumber : hourNumber + 12
+      date.setUTCHours(hour24Clock, parseInt(minutes), parseInt(seconds))
+    } else {
+      date.setUTCHours(0, 0, 0, 0)
+    }
+
+    return date
+  })()
+
+  // create applicant
+  const applicant = ((): ApplicantUpdate => {
+    const phoneNumber: string | null = data?.phoneNumber || null
+    const { applicant: applicantData } = data.application
+    const phoneNumberType: string | null = applicantData.phoneNumberType || null
+    const noEmail = !applicantData.emailAddress
+    const noPhone = !phoneNumber
+    const workInRegion: string | null = applicantData?.workInRegion || null
+    const emailAddress: string | null = applicantData?.emailAddress || null
+
+    const workAddress = getAddress(
+      applicantData?.workInRegion === "yes",
+      applicantData?.workAddress
+    )
+
+    return {
+      ...applicantData,
+      ...data.dateOfBirth,
+      emailAddress,
+      workInRegion,
+      workAddress,
+      phoneNumber,
+      phoneNumberType,
+      noEmail,
+      noPhone,
+    }
+  })()
+
+  // additional phone
+  const {
+    additionalPhoneNumber: additionalPhoneNumberData,
+    additionalPhoneNumberType: additionalPhoneNumberTypeData,
+    mailingAddress: mailingAddressData,
+    additionalPhoneNumber,
+    contactPreferences,
+    sendMailToMailingAddress,
+    alternateContact,
+    accessibility,
+    preferences,
+    demographics,
+    preferredUnit,
+  } = data.application
+
+  const additionalPhone = !additionalPhoneNumberData
+  const additionalPhoneNumberType = additionalPhoneNumberTypeData
+    ? additionalPhoneNumberTypeData
+    : null
+
+  const mailingAddress = getAddress(sendMailToMailingAddress, mailingAddressData)
+
+  // pass blank address, not used for now everywhere
+  const alternateAddress = getAddress(false, null)
+
+  const { incomeMonth, incomeYear, householdMembers } = data
+
+  const incomePeriod: IncomePeriod | null = data.application?.incomePeriod || null
+
+  const income = incomePeriod === IncomePeriod.perMonth ? incomeMonth : incomeYear || null
+  const incomeVouchers =
+    data.application.incomeVouchers === "yes"
+      ? true
+      : data.application.incomeVouchers === "no"
+      ? false
+      : null
+
+  const acceptedTerms =
```

Comment:
Please add type `bool | null`

Reply:
I think it's correct while we have type inference :)